File: 02. Recursion/04. pure_recursion.py
def collectOddValues(arr):
    newArr = []

    if (len(arr) == 0):
        return newArr

    if (arr[0] % 2 != 0):
        newArr.append(arr[0])

    arr.pop(0)
    newArr += collectOddValues(arr)
    return newArr


print(collectOddValues([11, 22, 4131, 1246, 9984, 3525]))
print(collectOddValues([3, 21, 531, 316, 984, 125]))

File: 02. Recursion/04. pure_recursion.js
function collectOddValues(arr) {
    let newArr = [];

    if (arr.length == 0)
        return newArr

    if (arr[0] % 2 != 0)
        newArr.push(arr[0])

    newArr = newArr.concat(collectOddValues(arr.splice(1)));
    return newArr;
}

console.log(collectOddValues([11, 22, 4131, 1246, 9984, 3525]))
console.log(collectOddValues([3, 21, 531, 316, 984, 125]))
